server/src/test/java/org/elasticsearch/transport/ConnectionManagerTests.java

```diff
@@ -124,11 +124,10 @@ public void onNodeDisconnected(DiscoveryNode node, Transport.Connection connecti
         assertEquals(1, nodeDisconnectedCount.get());
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/49903")
     public void testConcurrentConnectsAndDisconnects() throws BrokenBarrierException, InterruptedException {
         DiscoveryNode node = new DiscoveryNode("", new TransportAddress(InetAddress.getLoopbackAddress(), 0), Version.CURRENT);
-        Transport.Connection connection = new TestConnect(node);
         doAnswer(invocationOnMock -> {
+            Transport.Connection connection = new TestConnect(node);
             ActionListener<Transport.Connection> listener = (ActionListener<Transport.Connection>) invocationOnMock.getArguments()[2];
             if (rarely()) {
```

Comment:
I think it would be good to have these three branches chosen with more equal probabilities (and similarly for the connection validator lambda). As it is, we only complete both listeners on the same thread one in every 10,000 runs I think.